### PR TITLE
DOCSP-32753: Update Kotlin's Define Object Model page

### PR DESCRIPTION
**Pull request by cbullinger**
## Pull Request Info

- Convert existing [Data Model](https://www.mongodb.com/docs/realm/sdk/kotlin/realm-database/schemas/) landing page to a container
- Move relevant content to [Define an Object Model](https://www.mongodb.com/docs/realm/sdk/kotlin/realm-database/schemas/define-realm-object-model/) page
- Update Define an Object Model page with any missing content, examples
- Clean up existing examples in kotlin test suite

### Jira

- https://jira.mongodb.org/browse/DOCSP-32753

### Staged Changes

- [Define an Object Model - Kotlin SDK](https://preview-mongodbcbullinger.gatsbyjs.io/realm/docsp-32753-define-object-model-page/sdk/kotlin/realm-database/schemas/define-realm-object-model/) - updated page
- [Model Data - Kotlin SDK](https://preview-mongodbcbullinger.gatsbyjs.io/realm/docsp-32753-define-object-model-page/sdk/kotlin/realm-database/schemas/) - updated; now a container

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/AsymmetricSyncTest.kt [hunk 1]
```diff
@@ -26,18 +26,6 @@ import kotlin.time.Duration.Companion.seconds
 // }
 class AsymmetricSyncTest : RealmTest() {
 
-    // :snippet-start: asymmetric-model
```
Comment (cbullinger) [inline examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/AsymmetricSyncTest.kt:29]: moved relevant object model definitions to new Schema.kt file

File: examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/DataTypes.kt [hunk 2]
```diff
@@ -170,6 +171,52 @@ class RealmSupportedTypes : RealmObject {
     var uuidOpt: RealmUUID? = null
     // :snippet-end:
 }
+
+// :snippet-start: timestamp-workaround
```
Comment (cbullinger) [inline examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/DataTypes.kt:175]: moved here to be with the other examples used on Supported Types page

File: examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/SchemaTest.kt [hunk 3]
```diff
@@ -79,24 +82,6 @@ class Cat: RealmObject {
     var _id: RealmUUID = RealmUUID.random()
 }
 
-// :snippet-start: example-schema
```
Comment (cbullinger) [inline examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/SchemaTest.kt:82]: deleted - not used anymore

File: examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/SchemaTest.kt [hunk 4]
```diff
@@ -126,55 +111,10 @@ class PersistedName_Cat : RealmObject {
...
-        // TODO This probably breaks at edge cases like MIN/MAX
-        RealmInstant.from(sec + 1, -1_000_000 + nano)
-    }
-}
-// :snippet-end:
 class SchemaTest: RealmTest() {
     @Test
     fun createUUIDTypes() {
         runBlocking {
-            // :snippet-start: open-with-class
```
Comment (cbullinger) [inline examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/SchemaTest.kt:177]: deleted snippet - not used anymore

File: source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt [hunk 5]
```diff
@@ -22,271 +28,223 @@ defining your Realm object model, refer to
 .. _kotlin-object-type:
 .. _kotlin-object-schema:
 
-Object Types & Schemas
-----------------------
+Realm Objects
+-------------
 
-Realm objects are regular Kotlin classes, and you can work with them 
+**Realm objects** are uniquely named Kotlin classes that you can work with
```
Comment (lindseymoore) [inline source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt:34]:
```suggestion
**Realm objects** are uniquely named instances of Kotlin classes that you can work with 
```
Think its more precise to say objects are instances of  a class. Or could say 'Realm object models' instead and get rid of the 'instances' part I suggested.

File: source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt [hunk 6]
```diff
@@ -22,271 +28,223 @@ defining your Realm object model, refer to
...
-The Realm Kotlin SDK supports to-one relationships, to-many relationships, 
-inverse relationships, and embedding objects within other objects. 
+- The :ref:`data type <kotlin-supported-types>` for each property. 
+  The Kotlin SDK supports the following data types: 
 
-For more information, refer to the :ref:`Relationships <kotlin-relationships>` 
-page.
+  - `Kotlin primitive types <https://kotlinlang.org/docs/basic-types.html>`__
+  - a limited subset of `BSON <https://bsonspec.org/>`__ types
+  - Realm-specific types, which you can use for unique
```
Comment (lindseymoore) [inline source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt:76]: Is there a page to link to for the Realm-specific types since you linked for the others?

File: source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt [hunk 7]
```diff
@@ -22,271 +28,223 @@ defining your Realm object model, refer to
...
+         constructor(): this("", 0) // Empty constructor required by Realm
+      }
 
-.. todo add link to Create when page is updated
+Define a Realm Object
+~~~~~~~~~~~~~~~~~~~~~
 
-Define a Collection 
--------------------
+To define a Realm object, create a Kotlin class that implements the `RealmObject
```
Comment (lindseymoore) [inline source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt:111]:
```suggestion
To define a Realm object model, create a Kotlin class that implements the `RealmObject 
```
Are you referring to creating an object instance or model here?

File: source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt [hunk 8]
```diff
@@ -22,271 +28,223 @@ defining your Realm object model, refer to
...
-.. literalinclude:: /examples/generated/kotlin/CreateTest.snippet.define-a-realm-set.kt
-   :language: kotlin
+.. tip:: 
+   
+   If you need to manually manage the lifecycle of a referenced object or 
+   want the referenced objects to persist after the deletion of the parent 
+   object, use a regular Realm object with a :ref:`to-one relationship 
+   <kotlin-to-one-relationship>` instead. 
+   For more information on when to use embedded relationships over non-embedded 
+   relationship, refer to :ref:`<kotlin-relationships>`.
```
Comment (lindseymoore) [inline source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt:148]:
```suggestion
   relationships, refer to :ref:`<kotlin-relationships>`.
```
Grammar fix

File: source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt [hunk 9]
```diff
@@ -22,271 +28,223 @@ defining your Realm object model, refer to
...
+
+To define an embedded object, create a Kotlin class that implements the
```
Comment (lindseymoore) [inline source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt:150]:
```suggestion
To define an embedded object type, create a Kotlin class that implements the 
```
if you're talking about creating a new object model.

File: source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt [hunk 10]
```diff
@@ -22,271 +28,223 @@ defining your Realm object model, refer to
...
+the Atlas Device Sync feature Data Ingest. For information on how to set up
+Data Ingest with your application, refer to :ref:`kotlin-stream-data-to-atlas`.
 
-Define an Embedded Object
--------------------------
+.. include:: /includes/kotlin-asymmetric-object.rst
 
-To define an :ref:`embedded object <kotlin-embedded-objects>`, derive a class from 
-`EmbeddedRealmObject <{+kotlin-local-prefix+}io.realm.kotlin.types/-embedded-realm-object/index.html>`__: 
+To define an :ref:`asymmetric object<kotlin-asymmetric-objects>`, create a
```
Comment (lindseymoore) [inline source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt:179]:
```suggestion
To define an :ref:`asymmetric object type<kotlin-asymmetric-objects>`, create a 
```
Could use 'type' or 'model' here, but perhaps be consistent about it throughout the page.

File: source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt [hunk 11]
```diff
@@ -22,271 +28,223 @@ defining your Realm object model, refer to
...
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _kotlin-define-realm-dictionary-type:
+
+Define a RealmDictionary/RealmMap
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The Realm Kotlin SDK does not support having a single primary constructor.
-The SDK requires an empty constructor to create objects. As a workaround,
-you can do something similar to the following:
+To define a :ref:`RealmDictionary <kotlin-realm-dictionary>`, specify its
```
Comment (lindseymoore) [inline source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt:239]:
```suggestion
To define a property as a :ref:`RealmDictionary <kotlin-realm-dictionary>`, specify its 
```
For consistency with other sections.

**Review by lindseymoore — CHANGES_REQUESTED** [review]
A few minor copy changes.

File: source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt (hunk 6 above)
Comment (cbullinger) [reply source/sdk/kotlin/realm-database/schemas/define-realm-object-model.txt:76]: it's technically the Supported Types pages linked in the intro statement for this list, but I can add a link to that specific section here.

**Review by lindseymoore — APPROVED** [review]
LGTM!
